torchx/specs/named_resources_config.py

```diff
@@ -0,0 +1,90 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+# All rights reserved.
+#
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree.
+
+# pyre-strict
+
+"""
+Configuration-based named resources that can be defined via .torchxconfig file.
+This allows users to define custom named resources with specific CPU, GPU, memory,
+and device requirements without hardcoding them.
+
+Example .torchxconfig:
+[named_resources]
+dynamic = {"cpu": 100, "gpu": 8, "memMB": 819200, "devices": {"vpc.amazonaws.com/efa": 1}}
+my_custom = {"cpu": 32, "gpu": 4, "memMB": 131072}
+"""
+
+import json
+import os
+from configparser import ConfigParser
+from typing import Callable, Dict, Mapping
+
+from torchx.specs.api import Resource
+
+
+def _load_config_file() -> ConfigParser:
+    """Load the .torchxconfig file from TORCHXCONFIG env var or current directory."""
+    config = ConfigParser()
+
+    # Check TORCHXCONFIG environment variable first, then current directory
+    config_path = os.environ.get("TORCHXCONFIG", ".torchxconfig")
+
+    if os.path.exists(config_path):
+        config.read(config_path)
+
+    return config
```

Comment:
we really don't want to create another way to read `.torchxconfig`. This should be done through `torchx.runner.config`. If we have a valid reason to have `torchx.specs` read `.torchxconfig` then we need to move `torchx.runner.config` to `torchx.config` or `torchx.specs.config`